### PR TITLE
Buble logic

### PR DESCRIPTION
**Pull request by Guillermo849**

### CODE REVIEW

File: service/validate_number.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+module ValidateNumber
```
Comment (OlekKovalevych) [inline service/validate_number.rb:3]: 'TODO: Move validate to private method'
'TODO: Fix convention of method name'
'TODO: Change method module and error name'

File: service/bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module BubbleSort
+  def self.sort(arr)
```
Comment (OlekKovalevych) [inline service/bubble_sort.rb:4]: 'TODO Remove recurring from sort method'
'TODO Rename module to SortAlgorithms, and sort method to bundle sort'
'TODO Use only each method'

**Review by OlekKovalevych — APPROVED** [review]

File: service/validate_number.rb (hunk 1 above)
Comment (Guillermo849) [reply service/validate_number.rb:3]: +1

File: service/bubble_sort.rb (hunk 2 above)
Comment (Guillermo849) [reply service/bubble_sort.rb:4]: +1

File: service/validation.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Validation
+  NotIntegerError = Class.new(StandardError)
```
Comment (OlekKovalevych) [inline service/validation.rb:4]: Could you please move this error to Runner class

File: service/validation.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+  private
+
+  def validate_integer(num)
```
Comment (OlekKovalevych) [inline service/validation.rb:8]:
```suggestion
  def valid_input_char?(num)
```

File: service/validation.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    return true if num == '0'
+    return true unless num.to_i.zero?
+
+    false
```
Comment (OlekKovalevych) [inline service/validation.rb:12]: Maybe in this way? 
```suggestion
       true unless num != '0' && num.to_i.zero?
```

**Review by OlekKovalevych — APPROVED** [review]
Could you please rename dir from ```service``` to ```services```

File: service/runner.rb [hunk 6]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require_relative './validation'
+require_relative './sort_algorithm'
+
+class Runner
+  include Validation
+  def run
```
[the remark this answers is not in the document]
Comment (Guillermo849) [reply service/runner.rb:8]: +1

File: service/validation.rb (hunk 3 above)
Comment (Guillermo849) [reply service/validation.rb:4]: +1

File: service/validation.rb (hunk 4 above)
Comment (Guillermo849) [reply service/validation.rb:8]: +1

File: service/validation.rb (hunk 5 above)
Comment (Guillermo849) [reply service/validation.rb:12]: +1

**Pull request merged**